Fig.Tests/StringConversionTests.cs

```diff
@@ -125,6 +125,133 @@ public void ConvertEnum()
+        {
+            var now = DateTime.Now;
+            Assert.AreEqual(new DateTime[] { now, now.AddDays(1) }, _converter.Convert<DateTime[]>(now.ToString("O") + "," + now.AddDays(1).ToString("O")));
+            Assert.AreEqual(new DateTime[] { now }, _converter.Convert<DateTime[]>(now.ToString("O")));
+            Assert.AreEqual(null, _converter.Convert<DateTime[]>(null));
+            Assert.That(() => _converter.Convert<DateTime[]>("1a"), Throws.Exception);
+        }
+
+        [Test]
+        public void ConvertDateTimeOffsetArray()
+        {
+            var now = DateTimeOffset.Now;
+            Assert.AreEqual(new DateTimeOffset[] { now, now.AddDays(1) }, _converter.Convert<DateTimeOffset[]>(now.ToString("O") + "," + now.AddDays(1).ToString("O")));
+            Assert.AreEqual(new DateTimeOffset[] { now }, _converter.Convert<DateTimeOffset[]>(now.ToString("O")));
+            Assert.AreEqual(null, _converter.Convert<DateTimeOffset[]>(null));
+            Assert.That(() => _converter.Convert<DateTimeOffset[]>("1a"), Throws.Exception);
+        }
+
+        [Test]
+        public void ConvertTimeSpanArray()
+        {
+            Assert.AreEqual(new TimeSpan[] { TimeSpan.FromMinutes(42), TimeSpan.FromMinutes(40) }, _converter.Convert<TimeSpan[]>(TimeSpan.FromMinutes(42).ToString() + "," + TimeSpan.FromMinutes(40).ToString()));
+            Assert.AreEqual(new TimeSpan[] { TimeSpan.FromMinutes(42) }, _converter.Convert<TimeSpan[]>(TimeSpan.FromMinutes(42).ToString()));
+            Assert.AreEqual(null, _converter.Convert<TimeSpan[]>(null));
+            Assert.That(() => _converter.Convert<TimeSpan[]>("1a"), Throws.Exception);
+        }
+
+        [Test]
+        public void ConvertUnsignedIntegerArray()
+        {
+            Assert.AreEqual(new uint[] { 1, 2 }, _converter.Convert<uint[]>("1,2"));
+            Assert.AreEqual(new uint[] { 1 }, _converter.Convert<uint[]>("1"));
+            Assert.AreEqual(null, _converter.Convert<uint[]>(null));
+            Assert.That(() => _converter.Convert<uint[]>("1a"), Throws.Exception);
+        }
+
+        [Test]
+        public void ConvertByteArray()
+        {
+            Assert.AreEqual(new byte[] { 1, 2 }, _converter.Convert<byte[]>("1,2"));
+            Assert.AreEqual(new byte[] { 1 }, _converter.Convert<byte[]>("1"));
+            Assert.AreEqual(null, _converter.Convert<byte[]>(null));
+            Assert.That(() => _converter.Convert<byte[]>("256"), Throws.Exception);
+        }
+
+        [Test]
+        public void ConvertFloatArray()
+        {
+            Assert.AreEqual(new float[] { 1, 2 }, _converter.Convert<float[]>("1,2"));
+            Assert.AreEqual(new float[] { 1 }, _converter.Convert<float[]>("1"));
+            Assert.AreEqual(null, _converter.Convert<float[]>(null));
+            Assert.That(() => _converter.Convert<float[]>("1a"), Throws.Exception);
+        }
+
+        [Test]
+        public void ConvertDoubleArray()
```

Comment:
Just a thought here going forward... In sweden we use "," as the decimal separator so we have a potential problem here with a double such as 3,14 yielding either [3.14] or [3.0, 14.0]

Reply:
Very good point, I hadn't even considered culture's for parsing could be a problem for the comma seperator. What would you suggest here?

Reply:
Well, I suppose we make to have proper decimal points in the config :) Just document that we only support InvariantCulture. For now at least

Reply:
I have added an xml comment on the method that it uses invariant culture to convert to array.